omnigibson/utils/transform_utils.py

```diff
@@ -344,207 +432,404 @@ def vec(values):
         values (n-array): a tuple of numbers
 
     Returns:
-        np.array: vector of given values
+        th.tensor: vector of given values
     """
-    return np.array(values, dtype=np.float32)
+    return th.tensor(values, dtype=th.float32)
 
 
-def mat4(array):
+@th.jit.script
+def mat4(tensor):
     """
-    Converts an array to 4x4 matrix.
+    Converts an tensor to 4x4 matrix.
 
     Args:
-        array (n-array): the array in form of vec, list, or tuple
+        tensor (n-tensor): the tensor in form of vec, list, or tuple
+
+    Returns:
+        th.tensor: a 4x4 th tensor
+    """
+    return th.tensor(tensor, dtype=th.float32).view((4, 4))
 
+
+@th.jit.script
+def quat2mat(quaternion):
+    """
+    Converts given quaternion to matrix.
+    Args:
+        quaternion (tensor): (..., 4) tensor where the final dim is (x,y,z,w) quaternion
     Returns:
```

Comment:
hmm I felt like I am just rubber-stamping these. Did we copy these from somewhere else, or did we actually implement these from scratch based on math formulas?

Reply:
write tests for these.